src/test.rs

```diff
@@ -704,3 +719,37 @@ mod inner {
         T16
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate as warp;
+    use warp::Filter;
+    use warp::Reply;
+
+    #[tokio::test]
+    async fn example_test() {
+        let filter = warp::post()
+            .and(warp::path::param::<u32>())
+            .and(warp::path::param::<u32>())
+            .map(|x: u32, y: u32| x + y)
+            // As we have Reply trait implemented for String, so turn u32 here to String so we can have some extra methods
+            .map(|reply: u32| {
+                reply
+                    .to_string()
+                    .with_status(warp::http::StatusCode::CREATED)
+            });
+
+        // Execute `sum` and get the `Extract` back.
+        let response = warp::test::request()
+            .method("POST")
+            .path("/1/2")
+            .filter(&filter)
+            .await
+            .unwrap();
+        assert_eq!(response.get_status(), &warp::http::StatusCode::CREATED);
+        assert_eq!(response.get_reply(), "3");
+
+        // Or simply test if a request matches (doesn't reject).
+        assert!(!warp::test::request().path("/1/-5").matches(&filter).await);
+    }
+}
```

Comment:
Just added an example test for users to learn about testing.

Reply:
Indeed, but that one has to be carried out with `#[tokio::main]`. Perhaps it is just nicer to have a "test" example here using `#[tokio::test]`?